distro/run/pom.xml

```
@@ -18,6 +18,12 @@
   <dependencyManagement>
     <dependencies>
 
+      <dependency>
+        <groupId>org.codehaus.groovy</groupId>
+        <artifactId>groovy-all</artifactId>
+        <version>${version.groovy}</version>
+      </dependency>
```

@tmetzke I realized that the Camunda Run `poms` don't inherit the `camunda-core-internal-dependencies` pom.xml.

We can replace this dependency with:

```suggestion
      <dependency>
        <groupId>org.camunda.bpm</groupId>
        <artifactId>camunda-core-internal-dependencies</artifactId>
        <version>${project.version}</version>
        <scope>import</scope>
        <type>pom</type>
      </dependency>
```

Is there any reason not to do this?

---

After adjusting it, I got this error when I run it:

```
Failed to instantiate [ch.qos.logback.classic.LoggerContext]
Reported exception:
java.lang.AbstractMethodError: ch.qos.logback.classic.pattern.EnsureExceptionHandling.process(Lch/qos/logback/core/Context;Lch/qos/logback/core/pattern/Converter;)V
        at ch.qos.logback.core.pattern.PatternLayoutBase.start(PatternLayoutBase.java:86)
        at ch.qos.logback.classic.encoder.PatternLayoutEncoder.start(PatternLayoutEncoder.java:28)
        at ch.qos.logback.classic.BasicConfigurator.configure(BasicConfigurator.java:50)
        at ch.qos.logback.classic.util.ContextInitializer.autoConfig(ContextInitializer.java:152)
        at org.slf4j.impl.StaticLoggerBinder.init(StaticLoggerBinder.java:85)
        at org.slf4j.impl.StaticLoggerBinder.<clinit>(StaticLoggerBinder.java:55)
        at org.slf4j.LoggerFactory.bind(LoggerFactory.java:150)
        at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:124)
        at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:417)
        at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:362)
        at org.apache.commons.logging.LogAdapter$Slf4jAdapter.createLocationAwareLog(LogAdapter.java:130)
        at org.apache.commons.logging.LogAdapter.createLog(LogAdapter.java:91)
        at org.apache.commons.logging.LogFactory.getLog(LogFactory.java:67)
        at org.apache.commons.logging.LogFactory.getLog(LogFactory.java:59)
        at org.springframework.boot.SpringApplication.<clinit>(SpringApplication.java:201)
        at org.camunda.bpm.run.CamundaBpmRun.main(CamundaBpmRun.java:25)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
        at org.springframework.boot.loader.PropertiesLauncher.main(PropertiesLauncher.java:467)
```

---

The core internal dependencies are meant for all non-Spring Boot-based modules, actually. They contain version management for a lot of dependencies that are otherwise defined by the Spring Boot BOM. If there are dependency versions that should be shared for all modules (Spring Boot and others), we define those in the `parent`.

---

Groovy is also part of Spring Boot's BOM, as far as I can see. So we could also decide to use that version. I assumed we want to fix it to that specific version we test with.

---

ah okay, then the current solution is the best one :)